js/src/index.parity.js

```diff
@@ -52,12 +53,34 @@ if (process.env.NODE_ENV === 'development') {
 }
 */
 
-const AUTH_HASH = '#/auth?';
-
+const TOKEN_QS = '?token=';
+const hashIndex = window.location.hash
+  ? window.location.hash.indexOf(TOKEN_QS)
+  : -1;
+const searchIndex = window.location.search
+  ? window.location.search.indexOf(TOKEN_QS)
+  : -1;
 let token = null;
 
-if (window.location.hash && window.location.hash.indexOf(AUTH_HASH) === 0) {
-  token = qs.parse(window.location.hash.substr(AUTH_HASH.length)).token;
+if (hashIndex !== -1) {
+  // extract from hash (e.g. http://127.0.0.1:8180/#/auth?token=...)
+  token = qs.parse(window.location.hash.substr(hashIndex + 1)).token;
```

Comment:
Should we do `try-catch` here in case `qs.parse` fails?

Reply:
Currently (looked at the code itself), doesn't have exceptions, but rather safe than sorry.